bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-layout.vue

```diff
@@ -0,0 +1,260 @@
+      <div
+        :style="{
+          height: 'inherit',
+          width: 'inherit',
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          backgroundImage:
+            'linear-gradient(-90deg, rgba(0, 0, 0, 0.1) 1px, transparent 1px), linear-gradient(rgba(0, 0, 0, 0.1) 1px, transparent 1px)',
+          backgroundSize: `${grid.pitch}px ${grid.pitch}px, ${grid.pitch}px ${grid.pitch}px`,
+          visibility: context.editmode && grid.enable ? 'inherit' : 'hidden',
+        }" />
+      <f7-button
+        v-if="!context.editmode"
+        @click="toggleFullscreen"
+        class="fullscreen-icon"
+        :icon-f7="
+          fullscreen
+            ? 'arrow_down_right_arrow_up_left'
+            : 'arrow_up_left_arrow_down_right'
+        " />
+      <div
+        v-if="context.editmode"
+        style="
+          opacity: 0.3;
+          padding: 4px;
+          position: absolute;
+          width: 100%;
+        ">
+        {{ getCurrentScreenResolution() }}
+        <span v-if="isRetina()"><f7-icon
+          tooltip="Screen resolution shown is the fullscreen resolution for websites. Real screen resolution is bigger."
+          f7="info_circle" /></span>
+      </div>
+      <oh-canvas-item
+        v-for="obj in layout"
+        :key="obj.id"
+        :id="obj.id"
+        :gridEnable="grid.enable"
+        :gridPitch="grid.pitch"
+        :context="childContext(obj.item)" />
+    </div>
+  </div>
+</template>
+
+<style lang="stylus">
+.oh-canvas-layout {
+  overflow: visible; // prevent widget menus from scrolling page
+
+  .oh-canvas-layout-container {
+    margin: auto; // center layout when smaller than current screen
+    background-color: var(--f7-page-bg-color); // theme bg (for fullscreen)
+    transform-origin: top;
+  }
+
+  .oh-canvas-background {
+    height: 100%;
+    width: 100%;
+    object-fit: contain;
+  }
+
+  .fullscreen-icon {
+    position: absolute;
+    top: 2px;
+    right: 2px;
+    z-index: 1000;
+  }
+}
+</style>
+
+<script>
+import mixin from '../widget-mixin'
+import OhCanvasItem from './oh-canvas-item'
+import { OhCanvasLayoutDefinition } from '@/assets/definitions/widgets/layout'
+
+export default {
+  mixins: [mixin],
+  widget: OhCanvasLayoutDefinition,
+  components: {
+    OhCanvasItem: OhCanvasItem
+  },
+  data () {
+    return {
+      layout: [],
+      screenWidth: Number,
+      screenHeight: Number,
+      fullscreen: this.$fullscreen.getState(),
+      navbarHidden: false,
+      style: {
+        width: Number,
+        height: Number,
+        scale: 1.0
+      },
+      grid: {
+        pitch: Number,
+        enable: false
+      }
+    }
+  },
+  created () {
+    if (this.config.layoutType === 'fixed' && this.config.fixedType === 'canvas') {
+      this.style.width = this.screenWidth = this.config.screenWidth || 1280
+      this.style.height = this.screenHeight = this.config.screenHeight || 720
+      this.grid.pitch = this.config.grid || 5
```

Comment:
In my opinion 5 is quite low, it's almost there's no grid - isn't something like 20 better, wdyt?

Reply:
I use the grid at 5 to space some widgets like buttons in areas of the canvas where they are not related to the canvas background image. In that case, it is useful for example to maintain a determined spacing between widgets (sort of UI guidelines). In any case, we can put a larger default. I raised to 20.